src/Tempest/Mapper/src/Casters/FileUploadCaster.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Mapper\Casters;
+
+use InvalidArgumentException;
+use Tempest\Http\Upload;
+use Tempest\Mapper\Caster;
+
+final readonly class FileUploadCaster implements Caster
```

Comment:
I don't see how this would work? The goal would be to map `\Psr\Http\Message\UploadedFileInterface` to `\Tempest\Http\Upload::__construct`.

On top of that, we're already doing that in `\Tempest\Http\Mappers\PsrRequestToRequestMapper:43`, so I don't think an additional caster is actually needed.